atc/exec/build/repository.go

```diff
@@ -1,71 +1,68 @@
-package artifact
+package build
 
 import (
 	"sync"
 
-	"github.com/concourse/concourse/atc/worker"
+	"github.com/concourse/concourse/atc/runtime"
 )
 
-// Name is just a string, with its own type to make interfaces using it
+// ArtifactName is just a string, with its own type to make interfaces using it
 // more self-documenting.
-type Name string
+type ArtifactName string
 
-// Repository is the mapping from a Name to an ArtifactSource.
+// Repository is the mapping from a ArtifactName to an Artifact.
 // Steps will both populate this map with new artifacts (e.g.  the resource
 // fetched by a Get step), and look up required artifacts (e.g.  the inputs
 // configured for a Task step).
 //
 // There is only one ArtifactRepository for the duration of a build plan's
 // execution.
 //
-// ArtifactRepository is, itself, an ArtifactSource. As an ArtifactSource it acts
-// as the set of all ArtifactSources it contains, as if they were each in
-// subdirectories corresponding to their ArtifactName.
 type Repository struct {
-	repo  map[Name]worker.ArtifactSource
+	repo  map[ArtifactName]runtime.Artifact
 	repoL sync.RWMutex
 }
 
 // NewArtifactRepository constructs a new repository.
 func NewRepository() *Repository {
 	return &Repository{
-		repo: make(map[Name]worker.ArtifactSource),
+		repo: make(map[ArtifactName]runtime.Artifact),
 	}
 }
 
-//go:generate counterfeiter . RegisterableSource
-// A RegisterableSource	artifact is an ArtifactSource which can be added to the registry
-type RegisterableSource interface {
-	worker.ArtifactSource
+//go:generate counterfeiter . RegisterableArtifact
+// A RegisterableArtifact is an Artifact which can be added to the registry
+type RegisterableArtifact interface {
+	runtime.Artifact
 }
 
 // RegisterSource inserts an ArtifactSource into the map under the given
```

Comment:
pls update this documentation snippet!

Reply:
I dont quite follow :(

Reply:
@xtreme-sameer-vohra RegristerArtifact not RegisterSource

Reply:
ah, i fooled myself with an outdated comment 😸